### DISCUSSION TITLE
Local dev environment

### DISCUSSION THREAD

**Discussion by cenwadike** [aurora-is-near/rainbow-bridge#731] [Q&A] [answered]
Can someone please direct me to appropriate resources or describe how to set up a development environment to run and test near and eth side of rainbow-bridge.
@mfornet

**Comment by mfornet** [comment, marked as answer]
You should take a look at: https://github.com/aurora-is-near/rainbow-bridge#deploying-and-using-locally

Let us know if you have any issue following those steps.

**Reply by cenwadike** [reply]
Thanks... I'll definitely reach out when I hit blockers.

**Comment by cenwadike** [comment]
Is there a similar guide to locally run https://github.com/aurora-is-near/rainbow-token-connector

**Reply by sept-en** [reply]
The guide is required for rainbow-bridge because it's about deploying multiple contracts in strict order and also running several infrastructure components (i.e. relayers). If we talk about `rainbow-token-connector` - there are multiple contracts in that repo so if you need to test any of those - you can just deploy appropriate smart contracts, no additional infrastructure is required. I agree some additional README for each of the components would be useful, this will be added later.
Until that, you can check [NEAR CLI docs](https://docs.near.org/docs/tools/near-cli) on how to use it for contract deployment, etc. For Solidity contracts - it's just a standard way of operating with those contracts, with no special requirements.

**Comment by cenwadike** [comment]
This may be a newb question. I tried running `./build.sh` in 

> bridge-token-factory 

after building 

> bridge-token

but got this error 
> `error: Set BRIDGE_TOKEN to be the path of the bridge token binary`.

I figured I need to set the url of `bridge_token.wasm` but have not successfully done that.

I tried adding this line of code (at different tries) 
> let BRIDGE_TOKEN = std::path::Path::new("../../res/bridge-token.wasm");
const BRIDGE_TOKEN_BINARY: &'static [u8] = include_bytes!(std::env!(....

> let BRIDGE_TOKEN: &str = "../../res/bridge-token.wasm";
const BRIDGE_TOKEN_BINARY: &'static [u8] = include_bytes!(std::env!(....

> let BRIDGE_TOKEN: &'static [u8] = "../../res/bridge-token.wasm";
const BRIDGE_TOKEN_BINARY: &'static [u8] = include_bytes!(std::env!(....

but kept getting different variant of this error
> `error: Set BRIDGE_TOKEN to be the path of the bridge token binary`

**Reply by mfornet** [reply]
Yes, those instructions are outdated, I'm going to updated them right now. Use [Makefile](https://github.com/aurora-is-near/rainbow-token-connector/blob/master/Makefile#L17) instead:

`make res/bridge_token_factory.wasm`

**Reply by mfornet** [reply]
Updated instructions here: https://github.com/aurora-is-near/rainbow-token-connector/pull/102

**Reply by cenwadike** [reply]
Thank you...it compiled successfully.
